hpx/util/batch_environments/pbs_environment.hpp

```diff
@@ -53,7 +31,7 @@ namespace hpx { namespace util { namespace batch_environments {
 
         std::size_t num_threads() const
         {
-            return num_threads_;
+            return std::size_t(-1);
```

Comment:
That's a breaking change, at least, I think we should rather change how the given nodelist is being interpreted. If you have a PBS job, and have a mppwidth=24 (that's what you want on supermic for example), you get each host repeated 24 times in the PBS_NODEFILE. To use that, you'd have to do some bash acrobatics to extract the number of nodes.